TASVideos.WikiEngine/MakeToc.cs

```diff
@@ -18,9 +18,18 @@ public static Element MakeToc(List<INode> document, int charStart)
 			.Cast<Element>()
 			.ToList();
 
-		var ret = new Element(charStart, "div") { Attributes = { ["class"] = "toc" } };
+		var ret = new Element(charStart, "div") { Attributes = { ["class"] = "card" } };
+		var header = new Element(charStart, "div") { Attributes = { ["class"] = "card-header" } };
+		var strong = new Element(charStart, "strong");
+		strong.Children.Add(new Text(charStart, "Table of contents"));
+		header.Children.Add(strong);
+		ret.Children.Add(header);
+		var body = new Element(charStart, "div") { Attributes = { ["class"] = "card-body" } };
+		ret.Children.Add(body);
 		var stack = new Stack<Element>();
 		stack.Push(ret);
+		stack.Push(header);
+		stack.Push(body);
```

Comment:
Does it make more sense to only add `body` to `stack`, not `header` or `ret`?

Reply:
I don't know how it's meant to work really. It's just what I managed to guess from the existing code: it pushes children to stack as new items after having added them to parents. Can you give a brief explanation? I couldn't even find similar stack usage anywhere else in wiki creation.

Reply:
It uses the stack to track what list element it's in as it traverses the TOC, so that when the TOC pops up a layer, it can pop up from one `<ul>` to the previous `<ul>` and start adding to that again.  The initial contents of the stack are relevant here as whatever is on the top of the stack when entering the loop will be the element that gets the top `<ul>` added to it.  But it doesn't pop up any layers beyond that start, so anything else in the stack isn't used.

If you're really interested, the best way to understand the algorithm might be to step through it with a debugger, writing down the contents of the stack at each step.  It's one of those cute algorithms that isn't really that cute, but is useful.  `private bool StartLineLists()` does some similar things; I think an algorithm like this is inevitable if you're trying to turn a one-dimensional list of "depth numbers" into a hierarchical element tree.